pkg/scheduler/framework/interface.go

```diff
@@ -127,6 +131,16 @@ func (s *Status) Message() string {
 	return strings.Join(s.reasons, ", ")
 }
 
+// SetFailedPlugin sets the given plugin name to s.failedPlugin.
+func (s *Status) SetFailedPlugin(plugin string) {
+	s.failedPlugin = plugin
```

Comment:
In theory yes, but in practice it is not that necessary because we don't call this on `nil` or a concrect `Success` status. (If deem needed, it's even more necessary to guard `AppendReason()` than this)

Reply:
ok, we can do that as a follow up.